src/App.tsx

```diff
@@ -1,155 +1,125 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import {
+  getTodos,
+  postTodo,
+  patchTodo,
+  deleteTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { Error } from './components/Errors';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosApp, setTodosApp] = useState<Todo[]>(todos);
```

Comment:
do you really need 2 states? maybe you can create variable `filteredTodos` by using `useMemo` and filter todos by filter there